src/core/tensor/tensor_math_cuda.h

```diff
@@ -324,12 +324,8 @@ void EltwiseMult<float, lang::Cuda>(const Tensor& in,
                                     const float x, Tensor* out, Context* ctx) {
   const float* inPtr = static_cast<const float*>(in.block()->data());
   float* outPtr = static_cast<float*>(out->block()->mutable_data());
-
-  float alpha = x, beta = 0.0;
-  check_cudnn(cudnnAddTensor(ctx->cudnn_handle,
-                             (void*)(&alpha), generate_tensor_nd_desc(in), inPtr,
-                             (void*)(&beta), generate_tensor_nd_desc(*out), outPtr
-                            ));
+  const size_t num = in.Size();
```

Comment:
is cuda::mult faster than cudnnAddTensor?

Reply:
Yes, must be. It is because cudnnAddTensor add two tensors alphaX+betaY and write the result to Y. Meanwhile, cuda::mult is for purely elementwise multiply. The computational cost of cudnnAddTensor is hence much higher than cuda::mult. We do not need to use a tensor add function to perform elementwise multiply purpose for the consideration of computation cost.

Reply:
In other word, we only need alphaX but not alphaX+betaY (while setting beta=0). The first case has a less computation cost than the latter.